zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceCancelProcessor.java

```diff
@@ -94,7 +94,8 @@ private boolean validateCommand(
         new AuthorizationRequest(
                 command,
                 AuthorizationResourceType.PROCESS_DEFINITION,
-                PermissionType.UPDATE_PROCESS_INSTANCE)
+                PermissionType.UPDATE_PROCESS_INSTANCE,
+                command.getValue().getTenantId())
```

Comment:
❌ This is wrong. When cancelling a process instance users don't specify a tenant id on the command. Only a process instance key. What we should do here is use the tenant id that's on this process instance:

`elementInstance.getValue().getTenantId()`

Reply:
Good catch 👍